service/src/test/java/uk/gov/hmcts/reform/fpl/service/orders/generator/C29RecoveryOfAChildDocumentParameterGeneratorTest.java

```diff
@@ -0,0 +1,303 @@
+                PlacedUnderOrder.EMERGENCY_PROTECTION_ORDER)
+                + getEntryMessage(selectedChildren.size())
+                + getInformMessage(caseData.getManageOrdersEventData())
+                + getProduceMessage()
+                + getRemoveMessage()
+                + getIsExparteMessage(true))
+            .orderHeader(ORDER_HEADER)
+            .orderMessage(ORDER_MESSAGE)
+            .build();
+
+        assertThat(generatedParameters).isEqualTo(expectedParameters);
+    }
+
+    private C29RecoveryOfAChildDocmosisParametersBuilder<?,?> expectedCommonParameters() {
+        return C29RecoveryOfAChildDocmosisParameters.builder()
+            .furtherDirections(FURTHER_DIRECTIONS)
+            .localAuthorityName(LA_NAME)
+            .dateOfIssue(FORMAT_LOCAL_DATE_TO_STRING)
+            .orderTitle(C29_RECOVERY_OF_A_CHILD.getTitle());
+    }
+
+    private String getStandardMessage(int numberOfChildren, PlacedUnderOrder order) {
+        String orderGrammar = order.equals(PlacedUnderOrder.EMERGENCY_PROTECTION_ORDER) ? "an " : "a ";
+
+        String childrenGrammar = numberOfChildren > 1 ? "children" : "child";
+        return format(
+            "The Court is satisfied that %s has parental responsibility for the %s by virtue of %s made on %s.%s",
+            LA_NAME,
+            childrenGrammar,
+            orderGrammar + order.getLabel(),
+            FORMAT_LOCAL_DATE_TO_STRING,
+            paragraphBreak);
+    }
+
+    private String getEntryMessage(int numberOfChildren) {
+        String childrenGrammar = numberOfChildren > 1 ? "children" : "child";
+        return format("The court authorises %s to enter the premises known as %s, "
+                + "and search for the %s, using reasonable force if necessary.%s",
+            OFFICER_NAME,
+            REMOVAL_ADDRESS.getAddressAsString(", "),
+            childrenGrammar,
+            paragraphBreak
+        );
+    }
+
+    private String getInformMessage(ManageOrdersEventData eventData) {
+        String policeAssignee = isNull(eventData.getManageOrdersOfficerName())
+            ? "a police constable"
+            : eventData.getManageOrdersOfficerName();
+        return format(
+            "The court requires any person who has information about where the child is, "
+            + "or may be, to give that information to %s or an officer of the court, "
+            + "if asked to do so.%s", policeAssignee, paragraphBreak
+        );
+    }
+
+    private String getProduceMessage() {
+        return format(
+            "The court directs that any person who can produce the child when asked to by %s to do so.%s",
+            OFFICER_NAME, paragraphBreak
+        );
+    }
+
+    private String getRemoveMessage() {
+        return format("The Court authorises %s to remove the child.%s", OFFICER_NAME, paragraphBreak);
+    }
+
+    private String getIsExparteMessage(boolean isExparte) {
```

Comment:
In a test, I'd personally prefer to have these as well-named constants. Having logic inside the test, in my opinion, makes it harder to read and more complex.

Reply:
I've updated it and split out now. 
My thought was that it was so small that the readability wasnt effected. However, im happy with changing it